File: bindings/java-ffm/src/main/java/org/maplibre/nativeffi/WrongThreadException.java
package org.maplibre.nativeffi;

/** Thrown when an owner-thread-affine native handle is used from the wrong thread. */
public final class WrongThreadException extends MapLibreException {
  public WrongThreadException(int nativeStatusCode, String diagnostic) {
    super(MapLibreStatus.WRONG_THREAD, nativeStatusCode, diagnostic);
  }
}

File: bindings/kotlin/src/commonMain/kotlin/org/maplibre/nativeffi/error/WrongThreadException.kt
package org.maplibre.nativeffi.error

/** Error for graphics-thread-affine native handles called from the wrong thread. */
public class WrongThreadException
internal constructor(nativeStatusCode: Int, diagnostic: String = "") :
  MaplibreException(MaplibreStatus.WRONG_THREAD, nativeStatusCode, diagnostic)
